src/screens/Settings/components/WineExtensions/index.tsx

```diff
@@ -0,0 +1,111 @@
+import { faSyncAlt } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import React, { useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { ToggleSwitch } from 'src/components/UI'
+import { ipcRenderer } from 'src/helpers'
+
+interface Props {
+  eacRuntime: boolean
+  battlEyeRuntime: boolean
+  gameMode: boolean
+  toggleEacRuntime: () => void
+  toggleBattlEyeRuntime: () => void
+  toggleGameMode: () => void
+}
+
+export default function WineExtensions({
+  eacRuntime,
+  battlEyeRuntime,
+  gameMode,
+  toggleEacRuntime,
+  toggleBattlEyeRuntime,
+  toggleGameMode
+}: Props) {
+  const [eacInstalling, setEacInstalling] = useState(false)
+  const [battlEyeInstalling, setBattlEyeInstalling] = useState(false)
+  const { t } = useTranslation()
+
+  const handleEacRuntime = async () => {
+    toggleEacRuntime()
+    if (!eacRuntime) {
+      // Check if GameMode is turned on if the EAC runtime was turned on
+      if (!gameMode) {
+        const { response } = await ipcRenderer.invoke('openMessageBox', {
+          buttons: [t('box.yes'), t('box.no')],
+          message: t(
+            'settings.eacRuntime.gameModeRequired.message',
+            'Feral GameMode is required for the EAC runtime to work correctly. Do you want to enable it now?'
+          ),
+          title: t(
+            'settings.eacRuntime.gameModeRequired.title',
+            'GameMode required'
+          )
+        })
+        if (response === 0) {
+          toggleGameMode()
+        }
+      }
+
+      const isInstalled = await ipcRenderer.invoke(
+        'isRuntimeInstalled',
+        'eac_runtime'
+      )
+      if (!isInstalled) {
+        setEacInstalling(true)
+        await ipcRenderer.invoke('downloadRuntime', 'eac_runtime')
```

Comment:
If you work with reject/ resolve you should catch and print out errors. Or you catch the errors directly in the backend.